### ISSUE TITLE
Customizable Keyboard Shortcuts

### ISSUE THREAD

**Issue by awsdert** [elementary/code#1431]
### Problem

Haven't found anywhere to do this yet so would appreciate this being added to the preferences dialogue. I'd like to be able to define my own shortcuts for supported actions, along with custom shell commands related to current file directory.

### Proposal

Create a new dialogue for them. As for the custom commands I think it would be best to make some custom environment variables that follow standard GNUmakefile rules can be accessed for them. So for example `gcc -o $* $<` assigned to something like F5 would evaluate to `cd "..." && gcc -o main main.c`

**Comment by awsdert** [comment]
I'm sure some users might want to launch from the base directory that was opened as a project instead so an option for that would be a good idea.

**Comment by jeremypw** [comment]
@danirabbit  I don't think customizable shortcuts of in-app actions are approved of in elementary in general? Its not something I want to support. 

However, ability to link shell commands to a hot-keys is a possibility although I am not sure whether the hot-keys should be user selectable (risking clashes)  or app specified (e.g. `<Alt>1, <Alt>2 ...` or chosen from a list linked to a hot-key.

**Comment by danirabbit** [comment]
Yeah I'm inclined to agree with Jeremy here. If you really feel you need to customize every shortcut, I think there are apps that offer that. But it is a large amount of work to build and maintain a system for customizing shortcuts (we know because we maintain one for system shortcuts).

I'm going to close this as "out of scope" and echo the sentiment that if there's a specific shortcut you think is missing or wrong or a specific missing feature please open individual reports for this. It's better to re-evaluate or add shortcuts and fix the defaults for everyone where we can before resorting to adding more options
